defender/staged/downloaded/Compound_Gasless_Voting.spec.js

```diff
@@ -0,0 +1,264 @@
+      body: batchedTxs,
+    },
+  };
+  return autotaskEvent;
+}
+
+function createCastVoteObject(address, proposalId, support, v, r, s) {
+  const voteObject = {
+    address,
+    proposalId,
+    support,
+    v,
+    r,
+    s,
+  };
+
+  return voteObject;
+}
+
+function createDelegateVoteObject(address, delegatee, nonce, expiry, v, r, s) {
+  const delegateVoteObject = {
+    address,
+    delegatee,
+    nonce,
+    expiry,
+    v,
+    r,
+    s,
+  };
+
+  return delegateVoteObject;
+}
+
+describe('check autotask', () => {
+  it('casts votes when given a valid signed message', async () => {
+    mockContract.getCurrentVotes = jest.fn().mockResolvedValue(100);
+    mockContract.state = jest.fn().mockResolvedValue(1);
+    const validVote1 = createCastVoteObject(
+      '0x1111111111111111111111111111111111111111',
+      0,
+      0,
+      '0x1b',
+      '0xda4429a9e8e6b54cb101b2df002039f2879ab4ca0e8fae64134942cb81f3e581',
+      '0x3b90a37dc078a82dfc418695b1d4473661aa4d24dd874ac68678894ff44a6b27',
+    );
+    const validVote2 = createCastVoteObject(
+      '0x2222222222222222222222222222222222222222',
+      0,
+      0,
+      '0x1b',
+      '0x2e04bc22ee95e9f1d77b8634a8f0bf68e73dc8dab0b531d7cce777878b2f00b0',
+      '0xd73d470d9b221117d96c88f26a57d89ae8470d986e8bac4398d8315f4cca1460',
+    );
+    const batchedTxs = [validVote1, validVote2];
+    const autotaskEvent = createAutotaskEvent(batchedTxs);
+    await handler(autotaskEvent);
+
+    expect(mockContract.castVoteBySig).toBeCalledTimes(2);
+    expect(mockContract.delegateBySig).toBeCalledTimes(0);
+    mockContract.castVoteBySig.mockClear();
+    mockContract.delegateBySig.mockClear();
+  });
+
+  it('does not cast votes the address has 0 delegated votes', async () => {
+    mockContract.getCurrentVotes = jest.fn().mockResolvedValue(0);
+    mockContract.state = jest.fn().mockResolvedValue(1);
+    const validVote1 = createCastVoteObject(
+      '0x1111111111111111111111111111111111111111',
+      0,
+      0,
+      '0x1b',
+      '0xda4429a9e8e6b54cb101b2df002039f2879ab4ca0e8fae64134942cb81f3e581',
+      '0x3b90a37dc078a82dfc418695b1d4473661aa4d24dd874ac68678894ff44a6b27',
+    );
+    const validVote2 = createCastVoteObject(
+      '0x2222222222222222222222222222222222222222',
+      0,
+      0,
+      '0x1b',
+      '0x2e04bc22ee95e9f1d77b8634a8f0bf68e73dc8dab0b531d7cce777878b2f00b0',
+      '0xd73d470d9b221117d96c88f26a57d89ae8470d986e8bac4398d8315f4cca1460',
+    );
+    const batchedTxs = [validVote1, validVote2];
+    const autotaskEvent = createAutotaskEvent(batchedTxs);
+    await handler(autotaskEvent);
+
+    expect(mockContract.castVoteBySig).toBeCalledTimes(0);
+    expect(mockContract.delegateBySig).toBeCalledTimes(0);
+    mockContract.castVoteBySig.mockClear();
+    mockContract.delegateBySig.mockClear();
+  });
+
+  it('does not cast votes if the proposal is not active', async () => {
+    mockContract.getCurrentVotes = jest.fn().mockResolvedValue(0);
```

Comment:
This test is currently passing, but not for the right reasons:
  - The return value of `mockContract.state` was set by the previous test case and it will have the same value here unless it is reset with `mockReset()` (see comment adding a `beforeEach`).
  - The return value for `mockContract.getCurrentVotes` is set to `0`, which means that `castVoteBySig` won't be called because there are no delegated votes.

To address these concerns, we'll set the state to a value other than `1` and we'll make the `getCurrentVotes` value something greater than `0`.
```suggestion
  it('does not cast votes if there are delegated votes but the proposal is not active', async () => {
    mockContract.state = jest.fn().mockResolvedValue(2);
    mockContract.getCurrentVotes = jest.fn().mockResolvedValue(100);
```